### PR TITLE
Add resource output abstraction and fs output implementation

### PR DESCRIPTION
**Pull request by wtam2018**
Add resource output abstraction and fs output implementation.   
This PR is the first part of https://issues.redhat.com/browse/GITOPS-93.

### CODE REVIEW

File: pkg/manifest/out/fs/filesystem_output.go [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+		return nil, err
+	}
+
+	return &output{
+		BaseOutput: out.New(),
+		rootDir:    outputFolder,
+	}, nil
+}
+
+// Write all outpout items to filesystem
```
Comment (bigkevmcd) [inline pkg/manifest/out/fs/filesystem_output.go]:
```suggestion
// Write all outpout items to filesystem
```
```suggestion
// Write all output items to filesystem.
```

File: pkg/manifest/out/fs/filesystem_output.go [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+		err := writeToFile(filepath.Join(o.rootDir, filename), item)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func writeToFile(filename string, item interface{}) error {
+	err := os.MkdirAll(filepath.Dir(filename), 0755)
```
Comment (bigkevmcd) [inline pkg/manifest/out/fs/filesystem_output.go]: How about we switch to using afero, and have this take an `afero.Fs`?

**Review by DhanashriPalodkar — APPROVED** [review]

**Review by bigkevmcd — CHANGES_REQUESTED** [review]
This is pulling in go-git but not using it?

File: pkg/manifest/out/fs/filesystem_output.go (hunk 2 above)
Comment (wtam2018) [reply pkg/manifest/out/fs/filesystem_output.go]: Sure.  Will do it in https://issues.redhat.com/browse/GITOPS-98.

Comment (wtam2018) [reply pkg/manifest/out/fs/filesystem_output.go]: switched to afero for the out/fs stuff but there are remaining methods (e.g. ioutils.IsExisting) to be converted which will be done in GITOPS98)

**Pull request closed without merging**